### PR TITLE
Fix `get_group` with categoricals

### LINKED ISSUE #9359: GroupBy get_group KeyError when category and npartitions >1
Appears there is an issue when using groupby get_group and the group was derived from a pandas categorical type. Only get error when npartitions is >1. Possibly related to #[5847](https://github.com/dask/dask/issues/5847)

````python
def checker(npartitions, toCategory):
    import dask.dataframe as dd
    import pandas as pd

    df = pd.DataFrame({'make':['a','b','c'], 'model': [2,5,7]})
    
    if toCategory:
        df['make'] = df['make'].astype('category')
        
    dd = dd.from_pandas(df, npartitions=npartitions)
    
    gbCol = 'make'
    for g in dd[gbCol].unique().compute():
        dd.groupby(gbCol).get_group(g).min(numeric_only=True).compute()
    return

checker(npartitions=2, toCategory=True)  # doesnt work, KeyError: 'a', 2 partitions
checker(npartitions=2, toCategory=False) # works
checker(npartitions=1, toCategory=True)  # works, only 1 partition
checker(npartitions=1, toCategory=False) # works
````
**Traceback**:
````python-traceback
Traceback (most recent call last):

  File "C:\Users\User1\AppData\Local\Temp\ipykernel_5608\2821702366.py", line 18, in <cell line: 18>
    checker(npartitions=2, toCategory=True)  # doesnt work, KeyError: 'a', 2 partitions

  File "C:\Users\User1\AppData\Local\Temp\ipykernel_5608\2821702366.py", line 15, in checker
    dd.groupby(gbCol).get_group(g).min(numeric_only=True).compute() #get their associated group which is dataframe type

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\base.py", line 315, in compute
    (result,) = compute(self, traverse=False, **kwargs)

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\base.py", line 598, in compute
    results = schedule(dsk, keys, **kwargs)

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\threaded.py", line 89, in get
    results = get_async(

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\local.py", line 511, in get_async
    raise_exception(exc, tb)

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\local.py", line 319, in reraise
    raise exc

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\local.py", line 224, in execute_task
    result = _execute_task(task, data)

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\optimization.py", line 990, in __call__
    return core.get(self.dsk, self.outkey, dict(zip(self.inkeys, args)))

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 149, in get
    result = _execute_task(task, cache)

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 119, in <genexpr>
    return func(*(_execute_task(a, cache) for a in args))

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 113, in _execute_task
    return [_execute_task(a, cache) for a in arg]

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 113, in <listcomp>
    return [_execute_task(a, cache) for a in arg]

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 119, in <genexpr>
    return func(*(_execute_task(a, cache) for a in args))

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 113, in _execute_task
    return [_execute_task(a, cache) for a in arg]

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 113, in <listcomp>
    return [_execute_task(a, cache) for a in arg]

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\utils.py", line 71, in apply
    return func(*args, **kwargs)

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\dataframe\core.py", line 6650, in apply_and_enforce
    df = func(*args, **kwargs)

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\dask\dataframe\groupby.py", line 239, in _groupby_get_group
    return grouped.get_group(get_key)

  File "C:\ProgramData\Anaconda3\envs\env3\lib\site-packages\pandas\core\groupby\groupby.py", line 747, in get_group
    raise KeyError(name)

KeyError: 'a'
````

**Environment**:

- Dask version: 2022.8.0
- Python version: 3.10
- Operating System: Windows 10
- Install method (conda, pip, source): conda

### PR DESCRIPTION
**Pull request by pavithraes**
- [x] Closes #9359
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

TODO before merging:
- open a new issue around raising a KeyError early if the group is not present

### CODE REVIEW

File: dask/dataframe/groupby.py [hunk 1]
```diff
@@ -1633,7 +1632,10 @@ def last(self, split_every=None, split_out=1):
             token="last", func=M.last, split_every=split_every, split_out=split_out
         )
 
-    @derived_from(pd.core.groupby.GroupBy)
+    @derived_from(
+        pd.core.groupby.GroupBy,
+        inconsistencies="If the group is not present, Dask will return an empty Series/DataFrame.",
```
Comment (pavithraes) [inline dask/dataframe/groupby.py:1637]: We didn't raise a `KeyError` if the group wasn't present even before this PR.

Comment (ian-r-rose) [reply dask/dataframe/groupby.py:1637]: Interesting, I'm surprised by that. Do you think we should?

Comment (pavithraes) [reply dask/dataframe/groupby.py:1637]: As we discussed, this might be out of scope for this PR. :)

**Review by ian-r-rose — APPROVED** [review]
Thanks @pavithraes!

**Pull request merged**